src/MatrixFields/lazy_operators.jl

```diff
@@ -0,0 +1,103 @@
+abstract type AbstractLazyOperator end
+
+abstract type AbstractLazyOperatorStyle <: Base.Broadcast.BroadcastStyle end
+struct LazyOperatorStyle <: AbstractLazyOperatorStyle end
+struct LazyOperatorStyleWithSpaceAvailable <: AbstractLazyOperatorStyle end
+
+Base.Broadcast.broadcastable(lazy_op::AbstractLazyOperator) = lazy_op
+
+Base.Broadcast.BroadcastStyle(::Type{<:AbstractLazyOperator}) =
```

Comment:
This is probably one source of invalidations, is it needed?

Reply:
I need to somehow specify that a broadcast expression with an `AbstractLazyOperator` should have the `LazyOperatorStyle`. This is the same pattern we use to specify the `BroadcastStyle` for `DataLayout`s, `Field`s, `FieldVector`s, and `Operator`s. So, if this is triggering invalidations, then all of the other `BroadcastStyle` methods in `ClimaCore` are similarly triggering invalidations. I don't think the `Broadcast` API provides an alternative way to specify this information at the moment.